cmdr/src/giti/branch/checkout.rs

```diff
@@ -0,0 +1,149 @@
+/*
+ *   Copyright (c) 2023 R3BL LLC
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+use std::process::Command;
+
+use r3bl_ansi_color::{AnsiStyledText, Color, Style};
+use r3bl_rs_utils_core::{log_debug, CommonResult};
+use r3bl_tuify::{select_from_list_with_multi_line_header,
+                 SelectionMode,
+                 StyleSheet,
+                 LIGHT_GRAY_COLOR,
+                 SUCCESS_COLOR};
+
+use super::{get_branches, try_get_current_branch};
+use crate::giti::{report_unknown_error_and_propagate, single_select_instruction_header};
+
+pub fn try_checkout_branch() -> CommonResult<()> {
+    let default_header_style = [
+        Style::Foreground(Color::Rgb(171, 204, 242)),
+        Style::Background(Color::Rgb(31, 36, 46)),
+    ];
+
+    let instructions_and_branches = {
+        let mut instructions_and_branches = single_select_instruction_header();
+        let header = AnsiStyledText {
+            text: "Select a branch to switch to:",
```

Comment:
At some point, we might want to take all the static text strings and put them in an enum (where color constants are defined, for eg) and then use the enums or constants here, rather than string literals.

Same pattern as color can be used here.

```rust
enum UIStrings { Message1, Message2 }
impl UIStrings{
  pub fn to_string(&self) -> String {
    match self {
      Message1 => "foo bar baz".to_string(), 
      // etc.
    }
  }
}
```

The enum above might go in a file called `ui_strings.rs` that goes in the giti folder (for giti as an example). For edi, there would be a `ui_strings.rs` file in the edi folder. For common `ui_strings.rs` maybe we can place these at the src/ folder (very similar to the pattern we might use with `color_constants.rs`.